### ISSUE TITLE
Use same data object for teal_data and in modules

### ISSUE THREAD

**Issue by nikolas-burkoff** [insightsengineering/teal.data#102]
Following on from https://github.com/insightsengineering/teal/issues/731
Can we also use `qenv` 

To be refined at some point @gogonzo

**Comment by gogonzo** [comment]
> - Can we also use `qenv` instead of CodeClass 

CodeClass has important feature which we can't find an alternative. In CodeClass one can extract code of particular datasets as it keeps "dataname" attribute with all code-elements. This is important in module_nested_tabs where module gets subset of datasets and code.

**Comment by nikolas-burkoff** [comment]
> CodeClass has important feature which we can't find an alternative. In CodeClass one can extract code of particular datasets as it keeps "dataname" attribute with all code-elements. This is important in module_nested_tabs where module gets subset of datasets and code.

Fair enough, so maybe CodeClass could be built on top of qenv rather than being completely independent etc.

**Comment by gogonzo** [comment]
> Fair enough, so maybe CodeClass could be built on top of qenv rather than being completely independent etc.

Yes, it's a one solution and another is having something what CodeDepends have where one can obtain the expressions which are dependent on some object and reverse.
